src/Microsoft.DocAsCode.Common/Loggers/HtmlLogListener.cs

```diff
@@ -0,0 +1,144 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.DocAsCode.Common
+{
+    using System;
+    using System.IO;
+    using System.Text.RegularExpressions;
+
+    using Newtonsoft.Json;
+
+    public sealed class HtmlLogListener : ILoggerListener
+    {
+        private readonly StreamWriter _writer;
+
+        private const LogLevel LogLevelThreshold = LogLevel.Diagnostic;
+
+        private static readonly Regex EscapeWithEncode = new Regex(@"&", RegexOptions.Compiled);
+        private static readonly Regex EscapeWithoutEncode = new Regex(@"&(?!#?\w+;)", RegexOptions.Compiled);
+
+#if !NetCore
+        public HtmlLogListener(string reportPath)
+        {
+            var dir = Path.GetDirectoryName(reportPath);
+            if (!string.IsNullOrEmpty(dir))
+            {
+                Directory.CreateDirectory(dir);
+            }
+            _writer = new StreamWriter(reportPath, true);
+            _writer.WriteLine("<html>");
+            _writer.WriteLine(@"<body lang=""en-us"">");
+            _writer.WriteLine(@"<h1 id=""azure_migration_report"">Azure Content Pre Migration Report</h1>");
+            _writer.WriteLine(@"<table border=""1"">");
+            _writer.WriteLine(@"<tr><td>Severity</td><td>Message</td><td>File</td><td>Line</td><td>Time</td></tr>");
+        }
+#endif
+
+        public HtmlLogListener(StreamWriter writer)
+        {
+            if (writer == null)
+            {
+                throw new ArgumentNullException(nameof(writer));
+            }
+            _writer = writer;
```

Comment:
Will add here

---
In reply to: [74730872](https://github.com/dotnet/docfx/pull/605#discussion_r74730872) [](ancestors = 74730872)